### PR TITLE
Task subscription api retry queue

### PR DESCRIPTION
**Pull request by zzstoatzz**
adds a retry queue for when tasks are submitted and there is no running `TaskServer`
image

empirically this solves the problem I described earlier:
- prefect server start
- run a TaskServer
- submit a task -> runs great :white-check-mark2:
- stop the TaskServer
- submit a task while no task server running
- start the TaskServer -> subscription created
- nothing gets picked up

### CODE REVIEW

File: src/prefect/server/api/task_runs.py [hunk 1]
```diff
@@ -289,10 +301,27 @@ async def scheduled_task_subscription(
...
+                await websocket.send_json(task_run.dict(json_compatible=True))
+                # Log or handle successful send if needed
+            except subscriptions.NORMAL_DISCONNECT_EXCEPTIONS:
+                # If sending fails, put the task back into the retry queue
+                await retry_queue.put(task_run)
+                break  # Exit the loop to handle the disconnection
 
     except subscriptions.NORMAL_DISCONNECT_EXCEPTIONS:
-        pass  # it's fine if a client disconnects either normally or abnormally
+        pass  # Handle normal disconnections
```
[the remark this answers is not in the document]
Comment (zzstoatzz) [reply src/prefect/server/api/task_runs.py]: good call - updated in [a47bb45](https://github.com/PrefectHQ/prefect/pull/11789/commits/a47bb451bd02ad251155072c10789177780064de)

File: src/prefect/server/api/task_runs.py [hunk 2]
```diff
@@ -37,16 +37,28 @@
...
+_scheduled_task_runs_queues: Dict[
+    asyncio.AbstractEventLoop, asyncio.Queue[schemas.core.TaskRun]
+] = {}
+_retry_task_runs_queues: Dict[
+    asyncio.AbstractEventLoop, asyncio.Queue[schemas.core.TaskRun]
+] = {}
 
 
-def scheduled_task_runs_queue() -> asyncio.Queue:
+def scheduled_task_runs_queue() -> asyncio.Queue[schemas.core.TaskRun]:
```
[the remark this answers is not in the document]
Comment (zzstoatzz) [reply src/prefect/server/api/task_runs.py]: doh [a47bb45](https://github.com/PrefectHQ/prefect/pull/11789/commits/a47bb451bd02ad251155072c10789177780064de)

**Comment by zzstoatzz** [comment]
hmm

> look into sending a ping and awaiting its pong before we accept that the task has been delivered

looks like starlette's `WebSocket` doesn't have these out of the box, so i took a stab at an ack here - let me know if we maybe should hold off on this

File: src/prefect/client/subscriptions.py [hunk 3]
```diff
@@ -33,7 +33,15 @@ async def __anext__(self) -> S:
...
             try:
                 await self._ensure_connected()
-                return self.model.parse_raw(await self._websocket.recv())
+                message = await self._websocket.recv()
+
+                message_data = orjson.loads(message)
+
+                if message_data.get("type") == "ping":
+                    await self._websocket.send(orjson.dumps({"type": "pong"}).decode())
+                    continue
```
Comment (chrisguidry) [inline src/prefect/client/subscriptions.py:42]: Oh bummer, I thought we could use the built-in `websocket.ping()` but that doesn't seem to be available in the server-side sockets.  This works great too!

**Review by chrisguidry — APPROVED** [review]

**Pull request merged**